base/math.jl

```diff
@@ -102,6 +102,80 @@ macro horner(x, p...)
     return Expr(:block, :(local t = $(esc(x))), ex, :r)
 end
 
+"""
+    evalpoly(x, p::Tuple)
+
+Evaluate the polynomial ``\\sum_k p[k] x^{k-1}`` for the coefficients `p[1]`, `p[2]`, ...;
+that is, the coefficients are given in ascending order by power of `x`. This function
+generates efficient code using Horner's method with loops unrolled at compile time.
+
+# Example
+```jldoctest
+julia> evalpoly(2, (1, 2, 3))
+17
+```
+"""
+function evalpoly(x, p::Tuple)
```

Comment:
shouldn't `z::Real`  (probably doesn't matter)?

Reply:
It was intentional, since Horner's rule should apply to any object for which multiplication and addition of scalars are defined.